platform/core.startup/src/org/netbeans/core/startup/TopLogging.java

```diff
@@ -309,6 +313,32 @@ private static String createBootClassPath() {
         return sb.toString();
     }
 
+    private List<String> createJavaBootModuleList() {
+        // TODO JDK 11 equivalent
+//        return ModuleLayer.boot().modules().stream()
+//                .map(Module::getName)
+//                .sorted()
+//                .collect(Collectors.toList());
+        try {
+            Class<?> mod_class = Class.forName("java.lang.Module");
+            Class<?> ml_class = Class.forName("java.lang.ModuleLayer");
+            @SuppressWarnings("unchecked")
+            Set<?> mods = (Set<?>)ml_class.getDeclaredMethod("modules").invoke(
+                ml_class.getDeclaredMethod("boot").invoke(null)
+            );
+            return mods.stream().map(mod -> {
+                try {
+                    return (String) mod_class.getMethod("getName").invoke(mod);
```

Comment:
You could pull the method lookup out of the loop.

Reply:
oh you are right! will update